lib/Yield/Assembly/LLVMDisassemblerInterface.cpp

```diff
@@ -394,7 +390,12 @@ DI::instruction(const MetaAddress &Where, llvm::ArrayRef<uint8_t> RawBytes) {
   auto [Instruction, Size] = disassemble(Where, RawBytes, *Disassembler);
   if (Instruction.has_value()) {
     revng_assert(Size != 0);
-    auto P = parse(*Instruction, Where, *Printer, *SubtargetInformation);
+    std::uint64_t PC = Where.asPC();
+    if (Where.type() == MetaAddressType::Code_x86
+        || Where.type() == MetaAddressType::Code_x86_64) {
```

Comment:
Actually, we probably have to rework this further. The llvm disassembler is super bad with this and we probably better off just doing `relative` -> `absolute` address conversion ourselves :sadparrot:

Reply:
I have no idea what the problem here is and cannot comment intelligently.
What's the problem here?

Reply:
> x86 uses the next instruction's address as PC.

What do you mean? Can you make a specific example (by specific I mean an assembly snippet with the problem)?

In general, if there are such situation, they need to be documented in the code, otherwise it seems like magic.

Reply:
This snippet is gone already on my working branch, I'll close this thread and then we can make a new one about the updated solution (there were other problems with such an ad-hoc way to deal with this).